### PR TITLE
Revert "Use Date.from(). (#7516)"

### PR DESCRIPTION
**Pull request by swankjesse**
This reverts commit 55c3013b81c3c985d2797ae7dc48401b1fe72ae9.

### CODE REVIEW

File: okhttp/src/jvmMain/kotlin/okhttp3/Headers.kt [hunk 1]
```diff
@@ -180,25 +184,35 @@ actual class Headers internal actual constructor(
...
+    }
 
     /**
      * Add a header with the specified name and formatted instant. Does validation of header names
      * and value.
      */
-    fun add(name: String, value: Instant) = add(name, Date.from(value))
+    @IgnoreJRERequirement
+    fun add(name: String, value: Instant) = apply {
+      add(name, Date(value.toEpochMilli()))
```
Comment (JakeWharton) [inline okhttp/src/jvmMain/kotlin/okhttp3/Headers.kt:197]: Date.from is available if you have already have access to Instant

**Review by JakeWharton — COMMENTED** [review]
Adding the annotations makes sense, but why change the implementation? Both Instant and Date.from are available on the same API level.

File: okhttp/src/jvmMain/kotlin/okhttp3/Headers.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/jvmMain/kotlin/okhttp3/Headers.kt:197]: Great point. So maybe this is not a problem at all.

(I assume Animal Sniffer is still broken, as the `@IgnoreJRERequirement` annotations should be required?)

Comment (JakeWharton) [reply okhttp/src/jvmMain/kotlin/okhttp3/Headers.kt:197]: Yep. It should be failing since the signature is set to API 21 or something.

**Comment by swankjesse** [comment]
Obsoleted by https://github.com/square/okhttp/pull/7672

**Pull request closed without merging**